plugin/web/extensions/location/hook.ts

```diff
@@ -0,0 +1,95 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import { useShowGoogleModal } from "./state";
+import { MouseEventData } from "./types";
+import { postMsg } from "./utils";
+
+export default () => {
+  const distances = [
+    1, 2, 3, 5, 10, 20, 30, 50, 100, 200, 300, 500, 1000, 2000, 3000, 5000, 10000, 20000, 30000,
+    50000, 100000, 200000, 300000, 500000, 1000000, 2000000, 3000000, 5000000, 10000000, 20000000,
+    30000000, 50000000,
+  ];
+  let pixelDistance = 0;
+
+  const [currentPoint, setCurrentPoint] = useState<MouseEventData>();
+  const [currentDistance, setCurrentDistance] = useState("0 Km");
+  const [showGoogleModal, setShowGoogleModal] = useShowGoogleModal();
+  const [showTerraineModal, setShowTerrainModal] = useShowGoogleModal();
+
+  const updateCurrentPoint = useCallback((mousedata: MouseEventData) => {
+    setCurrentPoint(mousedata);
+  }, []);
+
+  const handlegoogleModalChange = useCallback(() => {
+    setShowGoogleModal(!showGoogleModal);
+    postMsg({ action: !showGoogleModal ? "modal-google-open" : "modal-close" });
+  }, [showGoogleModal]);
+  const handleTerrainModalChange = useCallback(() => {
+    setShowTerrainModal(!showTerraineModal);
+    postMsg({ action: !showTerraineModal ? "modal-Terrain-open" : "modal-close" });
+  }, [showTerraineModal]);
+
+  useEffect(() => {
+    (globalThis as any).addEventListener("message", (e: any) => {
+      if (e.source !== parent) return;
+      if (e.data.type) {
+        if (e.data.type === "mousedata") {
+          console.log(e.data);
+          updateCurrentPoint(e.data);
+        }
+        if (e.data.type === "getLocations") {
+          if (e.data.payload.point1 && e.data.payload.point2) {
+            pixelDistance =
+              getDistanceFromLatLonInKm(
+                e.data.payload.point1.lat,
+                e.data.payload.point1.lng,
+                e.data.payload.point2.lat,
+                e.data.payload.point2.lng,
+              ) * 1000;
+          }
+        }
+      }
+    });
+  });
```

Comment:
Be sure to add the `, []` at the end of the useEffect or else it will run on every re-render (https://stackoverflow.com/a/58579462/13699142)

Also, I think you need to do as I did in the Sidebar and removeEventListener on demounting or else every time the component re-renders a new addEventListener will be created. I can explain more tomorrow if its not clear what I mean

Reply:
Done - regarding dependency array , but need more explanation why to move the event Listener please

Reply:
f2279696fd7da46ec18feafdf3236b92f2a9fa0e

Reply:
for eventListener - Done - 756322989fa9a1d1facaf76a678d211c363fead7